packages/minifier-css/minifier.js

```diff
@@ -1,112 +1,209 @@
-var cssParse = Npm.require('css-parse');
-var cssStringify = Npm.require('css-stringify');
-var url = Npm.require('url');
-var path = Npm.require('path');
-
-CssTools = {
-  parseCss: cssParse,
-  stringifyCss: cssStringify,
-  minifyCss: function (cssText) {
-    return CssTools.minifyCssAst(cssParse(cssText));
+import path from 'path';
+import url from 'url';
+import Future from 'fibers/future';
+import postcss from 'postcss';
+import cssnano from 'cssnano';
+
+const CssTools = {
+  /**
+   * Parse the incoming CSS string; return a CSS AST.
+   *
+   * @param {string} cssText The CSS string to be parsed.
+   * @param {Object} options Options to pass to the PostCSS parser.
+   * @return {postcss#Root} PostCSS Root AST.
+   */
+  parseCss(cssText, options = {}) {
+    // This function previously used the `css-parse` npm package, which
+    // set the name of the css file being pased using  { source: 'filename' }.
+    // If included, we'll convert this to the `postcss` equivalent, to maintain
+    // backwards compatibility.
+    if (options.source) {
+      options.from = options.source;
+      delete options.source;
+    }
+    return postcss().process(cssText, options).root;
   },
-  minifyCssAst: function (cssAst) {
-    return MinifyAst(cssAst);
+
+  /**
+   * Using the incoming CSS AST, create and return a new object with the
+   * generated CSS string, and optional sourcemap details.
+   *
+   * @param {postcss#Root} cssAst PostCSS Root AST.
+   * @param {Object} options Options to pass to the PostCSS parser.
+   * @return {Object} Format: { code: 'css string', map: 'sourcemap deatils' }.
+   */
+  stringifyCss(cssAst, options = {}) {
+    // This function previously used the `css-stringify` npm package, which
+    // controlled sourcemap generation by passing in { sourcemap: true }.
+    // If included, we'll convert this to the `postcss` equivalent, to maintain
+    // backwards compatibility.
+    if (options.sourcemap) {
+      options.map = {
+        inline: false,
+        annotation: false,
+        sourcesContent: false,
+      };
+      delete options.sourcemap;
+    }
+
+    const f = new Future;
+    postcss().process(cssAst, options).then(result => {
+      f.return(result);
+    }).catch(error => {
+      f.throw(error);
+    });
+    transformResult = f.wait();
+    return {
+      code: transformResult.css,
+      map: transformResult.map ? transformResult.map.toJSON() : null,
+    };
+  },
+
+  /**
+   * Minify the passed in CSS string.
+   *
+   * @param {string} cssText CSS string to minify.
+   * @return {String[]} Array containing the minified CSS.
+   */
+  minifyCss(cssText) {
+    const f = new Future;
+    postcss([ cssnano ]).process(cssText).then(result => {
```

Comment:
Can the object returned from `postcss(...)` be saved for later, or is it important to create a new one each time?

Reply:
`postcss()` is just a helper function that creates and returns a new `Processor` object. So it's essentially the same as calling `new Processor()`. We can definitely save the returned `Processor` object. It's pretty lightweight so we might not gain much, and we'll have to adjust how  `CssTools.minifyCss` is passing the `cssnano` plugin into `postcss()`, but it's definitely doable. I'll adjust - thanks!

Reply:
I looked into this further @benjamn - unfortunately we can't share the object. In `CssTools` we're making 3 separate `postcss()` calls, one of which looks like `postcss([ cssnano ])`. So here we're passing a plugin into a `new Processor` call behind the scenes. If we were to change things around so we can share the `postcss()` created `Processor` object, I thought we could leverage the `Processor` objects [`use`](http://api.postcss.org/Processor.html#use) method, so we could essentially have something like `this._cssProcessor.process()` or `this._cssProcessor.use(cssnano).process()` (where `this._cssProcessor` points to our shared `Processor` object). It turns out though that the `use` method just adds the plugin to the object, so it's then used with any processing from that point forward (with or without the `use` method). Since we don't always want the plugin, I think our best bet is to leave this as is. Again, the created `Processor` object is pretty lightweight so this should be okay.